tests/functions/test_visualizations.py

```diff
@@ -65,6 +66,234 @@ def test_loading_lazily(monkeypatch, metadata_df):
     get_blob_as_stream.assert_not_called()
 
 
+def test_cytof_antibody_metadata_end_to_end(monkeypatch, metadata_df):
+    """Test addition of antibody metadata for cytof files"""
+    # Mock an CyTOF downloadable file record
+    cytof_record = MagicMock()
+    cytof_record.object_url = "foo"
+    cytof_record.upload_type = "cytof"
+    cytof_record.additional_metadata = {"foo": "bar"}
+    get_by_object_url = MagicMock()
+    get_by_object_url.return_value = cytof_record
+    monkeypatch.setattr(DownloadableFiles, "get_by_object_url", get_by_object_url)
+
+    # Mock GCS call
```

Comment:
This isn't mocking a call to Google Cloud Storage.

Reply:
fair point, copy-paste error